steps/generate_dataset_from_htme.py

```diff
@@ -223,64 +254,62 @@ def persist_json(json_location, values):
 def get_collection(collection_name):
     return (
         collection_name.replace("db.", "", 1)
-        .replace(".", "_")
-        .replace("-", "_")
-        .lower()
+            .replace(".", "_")
+            .replace("-", "_")
+            .lower()
     )
 
 
-def get_published_db_name():
-    published_database_name = "${published_db}"
-    return published_database_name
-
-
 def get_collections(secrets_response):
     try:
         collections = secrets_response["collections_all"]
         collections = {k.lower(): v.lower() for k, v in collections.items()}
-    except:
-        logging.error("Problem with collections list")
+    except BaseException as ex:
+        logging.error("Problem with collections list, stopping Spark" + str(ex))
+        spark.stop()
+        raise ex
     return collections
 
 
-def create_hive_on_published(json_location, collection_name):
-    hive_table_name = get_collection(collection_name)
-    src_hive_table = published_database_name + "." + hive_table_name
-    # print(f'src_hive_table ======={hive_table_name}')
-    the_logger.info("Creating Hive tables  : " + src_hive_table)
-    src_hive_drop_query = "DROP TABLE IF EXISTS %s" % src_hive_table
-    src_hive_create_query = (
-        """CREATE EXTERNAL TABLE IF NOT EXISTS %s(val STRING) STORED AS TEXTFILE LOCATION "%s" """
-        % (src_hive_table, json_location)
-    )
-    spark.sql(src_hive_drop_query)
-    spark.sql(src_hive_create_query)
-    return None
+def create_hive_tables_on_published(spark, all_processed_collections, published_database_name):
+    for (collection_name, collection_json_location) in all_processed_collections:
```

Comment:
How do we recover if something goes wrong here ,  processing all collections just because we can create hive tables isn't a good idea , may be we should extract this as a step so that we can rerun just that step ?